katika/settings.py

```diff
@@ -151,29 +152,54 @@
 # DATABASES #
 #############
 
+# DATABASES = {
+#     "default": {
+#         # Ends with "postgresql_psycopg2", "mysql", "sqlite3" or "oracle".
+#         #"ENGINE": "",
+#         # "ENGINE": "django.db.backends.sqlite3",
+#         #"ENGINE": "django.contrib.gis.db.backends.spatialite",
+#         "ENGINE": "django.contrib.gis.db.backends.postgis",
+#         # DB name or path to database file if using sqlite3.
+#         # "NAME": "dev.db",
+#         "NAME": "katika",
+#         #https://stackoverflow.com/questions/36214127/django-db-utils-operationalerror-fe-sendauth-no-password-supplied
+#         #"NAME": "",
+#         # Not used with sqlite3.
+#         # "USER": "postgres",
+#         "USER": "vagrant",
+#         # Not used with sqlite3.
+#         #"PASSWORD": "postgres",
+#         # Set to empty string for localhost. Not used with sqlite3.
+#         #"HOST": "localhost",
+#         # Set to empty string for default. Not used with sqlite3.
+#         #"PORT": "",
+#     }
+# }
+###################
+## CONFIGURABLE DATABASE SETTINGS
 DATABASES = {
-    "default": {
-        # Ends with "postgresql_psycopg2", "mysql", "sqlite3" or "oracle".
-        #"ENGINE": "",
-        # "ENGINE": "django.db.backends.sqlite3",
-        #"ENGINE": "django.contrib.gis.db.backends.spatialite",
-        "ENGINE": "django.contrib.gis.db.backends.postgis",
-        # DB name or path to database file if using sqlite3.
-        # "NAME": "dev.db",
-        "NAME": "katika",
-        #https://stackoverflow.com/questions/36214127/django-db-utils-operationalerror-fe-sendauth-no-password-supplied
-        #"NAME": "",
-        # Not used with sqlite3.
-        # "USER": "postgres",
-        "USER": "vagrant",
-        # Not used with sqlite3.
-        #"PASSWORD": "postgres",
-        # Set to empty string for localhost. Not used with sqlite3.
-        #"HOST": "localhost",
-        # Set to empty string for default. Not used with sqlite3.
-        #"PORT": "",
-    }
+    'dev': {
+        'ENGINE': 'django.db.backends.sqlite3',
+        'NAME': os.path.join(BASE_DIR, 'katika.sqlite3'),
+    },
+    'production': {
+	'ENGINE':  os.environ['KATIKA_DATABASE_ENGINE'],
```

Comment:
This parametrization is actually not necessary and a bit counterproductive. Local parameters can be added to `local_settings.py` and not pushed to the git repo.  
Having those parameters forces to have these environment parameter set. It is not necessary

Reply:
> @ntsieyo I've spent some time on this yesterday and realized eg. Django4.2 requires to upgrade Python as well. I did that but got stuck with `ModuleNotFoundError: No module named 'crispy_bootstrap3'` I knew such a big drop would take more time to check and I'd just like to let you know that I'm working on it whenever possible

did you run pip install -r requirements2.txt ?

Reply:
> > @ntsieyo I've spent some time on this yesterday and realized eg. Django4.2 requires to upgrade Python as well. I did that but got stuck with `ModuleNotFoundError: No module named 'crispy_bootstrap3'` I knew such a big drop would take more time to check and I'd just like to let you know that I'm working on it whenever possible
> 
> did you run pip install -r requirements2.txt ?

`requirements.txt` or `requirements2.txt`. We only used the first so far and didn't see the latter (in your pull-request)

Reply:
Ok I see. You have defined the crispy_boostrap stuff in requirements2.txt. so without installing those dependencies the code could not start, it was broken. What I did was to bump the version of these packages. This is one of the reasons why I updated base code to recent version because it was not possible to start the project locally. Some old version was not available.

Reply:
You don't have this problem in your server because you already have the old dependencies installed. You need to try to install your current master branch on a fresh system to see those issues.

Reply:
It could be but katika/requirements2.txt is probably some old left over file which shouldn't be relevant. That issue (crispy_form...) should be fixed separately  
I've installed those dependencies but now getting a 'geo_db_type' when running `python manage.py migrate`  

I'm really sorry to say but this is not efficient. Please @ntsieyo break this PR into separate (incremental) PRs otherwise it's going to take even more time

Reply:
Thanks @ntsieyo please take your time, no stress at all. We could have one for Python upgrade and another one for Django. The rest can be taken later.   
Local environment variables can be put in `local_settings.py` as I mentioned and no need to commit to the repo   
I'll remove `requirements2.txt` so to avoid confusion as well